newIDE/app/src/Profile/EmailVerificationDialog.js

```diff
@@ -0,0 +1,136 @@
+  sendEmailAutomatically: boolean,
+  showSendEmailButton: boolean,
+  onSendEmail: () => Promise<void>,
+|};
+
+export default function EmailVerificationDialog({
+  onClose,
+  authenticatedUser,
+  sendEmailAutomatically,
+  showSendEmailButton,
+  onSendEmail,
+}: Props) {
+  const isVerified =
+    !!authenticatedUser.firebaseUser &&
+    !!authenticatedUser.firebaseUser.emailVerified;
+
+  const [hasSentEmailManually, setHasSentEmailManually] = React.useState(false);
+
+  // Send the email once on dialog opening if configured as so.
+  React.useEffect(
+    () => {
+      if (!isVerified && sendEmailAutomatically) {
+        onSendEmail();
+      }
+    },
+    [isVerified, onSendEmail, sendEmailAutomatically]
+  );
+
+  // Check every 5 seconds if the email has been verified.
+  useInterval(
+    () => {
+      authenticatedUser.onRefreshFirebaseProfile().catch(() => {
+        // Ignore any error, will be retried anyway.
+      });
+    },
+    isVerified ? null : 5000
+  );
+
+  return (
+    <Dialog
+      title={null} // This dialog has a custom design to be more welcoming, the title is set in the content.
+      actions={[
+        isVerified ? (
+          <DialogPrimaryButton
+            label={<Trans>Done!</Trans>}
+            key="close"
+            primary
+            onClick={onClose}
+          />
+        ) : (
+          <FlatButton
+            label={<Trans>I'll do it later</Trans>}
+            key="close"
+            primary={false}
+            onClick={onClose}
+          />
+        ),
+      ]}
+      maxWidth="sm"
+      open
+      onRequestClose={onClose}
+      onApply={onClose}
+    >
+      <ColumnStackLayout
+        noMargin
+        expand
+        justifyContent="center"
+        alignItems="center"
+      >
+        <GDevelopGLogo fontSize="large" />
+        {!authenticatedUser.firebaseUser ? null : !authenticatedUser
+            .firebaseUser.emailVerified ? (
+          <ColumnStackLayout noMargin alignItems="center">
+            <Text size="title" align="center">
+              <Trans>Confirm your email</Trans>
+            </Text>
+            <Text align="center">
+              <Trans>
+                You will get access to special discounts on the GDevelop asset
+                store, as well as weekly stats about your games.
+              </Trans>
+            </Text>
+            {!hasSentEmailManually && showSendEmailButton ? (
+              <RaisedButton
+                primary
+                label={<Trans>Send it again</Trans>}
+                onClick={() => {
+                  setHasSentEmailManually(true);
+                  onSendEmail();
+                }}
+              />
+            ) : (
+              <BackgroundText>
+                <LineStackLayout justifyContent="center" alignItems="center">
+                  <CircularProgress size={20} />
+                  <Text>
+                    <Trans>
+                      Waiting for the email verification of
```

Comment:
I don't know if the english is right here, "A verification email has been sent to {}, waiting for validation". But I'm not sure it's better